python/python/lancedb/embeddings/cohere.py

```diff
@@ -61,14 +89,39 @@ class TextModel(LanceModel):
     """
 
     name: str = "embed-multilingual-v2.0"
+    source_input_type: str = "search_document"
+    query_input_type: str = "search_query"
     client: ClassVar = None
 
     def ndims(self):
         # TODO: fix hardcoding
-        return 768
+        if self.name in [
+            "embed-english-v3.0",
+            "embed-multilingual-v3.0",
+            "embed-english-light-v2.0",
+        ]:
+            return 1024
+        elif self.name in ["embed-english-light-v3.0", "embed-multilingual-light-v3.0"]:
+            return 384
+        elif self.name == "embed-english-v2.0":
+            return 4096
+        elif self.name == "embed-multilingual-v2.0":
+            return 768
+        else:
+            raise ValueError(f"Model {self.name} not supported")
+
+    def compute_query_embeddings(self, query: str, *args, **kwargs) -> List[np.array]:
+        return self.compute_source_embeddings(query, input_type=self.query_input_type)
```

Comment:
[Same as above] Non-blocking: Should this use the same `kwargs.get("input_type") or self.query_input_type` logic as in `compute_source_embeddings`? 
This logic is perfectly fine for the typical search/retrieval case. I'm much less familiar with the other use cases for vector databases but I would imagine other input types (such as `"clustering"`) might be useful.

Reply:
Yeah these are tested in slow tests [here](https://github.com/lancedb/lancedb/blob/main/python/python/tests/test_embeddings_slow.py) .
These functions are called internally with the right params, so you can set both source and query input types of your choice. Like this:
```
import lancedb
from lancedb.embeddings import get_registry

cohere = get_registry.get("cohere").create(name="model_name", source_input_type="classification", query_input_type="clustering")
```

See docs here- https://lancedb.github.io/lancedb/embeddings/default_embedding_functions/#ollama-embeddings

I realized we missed adding cohere docs, which I'm addressing in another PR, but you can see all accepted args that you can pass here - https://github.com/lancedb/lancedb/pull/1326/files#diff-e4f8e4ca897425fa2aa42069418ab1b10a35b3b8233e30ab278d776ffb96fddcR385

Reply:
Ahh ok, I'm following. Yea, what you have for setting the input types when creating the embedding function is even better. Thanks for the explanation!